### PR TITLE
MacOS: make mute sound actually work

### LINKED ISSUE #14100: (macOS) Muting has no effect: There are still sounds
### Minetest version

```true
minetest 5.8.0
```

### Active renderer

_No response_

### Irrlicht device

X11

### Operating system and version

macOS Ventura

### CPU model

Apple M1

### GPU model

_No response_

### OpenGL version

_No response_

### Summary

Even after pressing the Mute key or adjusting the sound to 0, the game is still making noises.

### Steps to reproduce

1. Create a world in whatever game
2. Click M for muting
3. It says Muted
4. Sounds were still made

### PR DESCRIPTION
**Pull request by sfence**
For some reason on macOS OpenAL looks to ignore setting global gain to 0.0f, but setting it to the smallest float normalized value takes effect.

Ready for Review.

## How to test

Run on macOS and try to mute/unmute the game.

### CODE REVIEW

File: src/client/sound.cpp [hunk 1]
```diff
@@ -101,7 +101,11 @@ void sound_volume_control(ISoundManager *sound_mgr, bool is_window_active)
 {
 	bool mute_sound = g_settings->getBool("mute_sound");
 	if (mute_sound) {
+#if defined(__APPLE__)
+		sound_mgr->setListenerGain(0.001f);
+#else
 		sound_mgr->setListenerGain(0.0f);
+#endif
```
Comment (grorp) [inline src/client/sound.cpp:108]: What if someone explicitly sets `sound_volume = 0`?

**Comment by Desour** [comment]
How low can you make it? Does `0.0000001f` also work for example?

Maybe add a check for 0.0f in `OpenALSoundManager::setListenerGain`.

File: src/client/sound.cpp (hunk 1 above)
Comment (sfence) [reply src/client/sound.cpp]: Moved to `OpenALSoundManager::setListenerGain`

**Comment by sfence** [comment]
> How low can you make it? Does `0.0000001f` also work for example?
> 
> Maybe add a check for 0.0f in `OpenALSoundManager::setListenerGain`.

Also, the minimum float value works. So I used it (normalized one).

**Comment by Desour** [comment]
Looks fine. But I can't test.

**Pull request merged**